src/metabase/driver/common/parameters/values.clj

```diff
@@ -116,6 +122,28 @@
                  (sequential? value-info-or-infos) (mapv #(dissoc % :target) value-info-or-infos))
                i/no-value)})))
 
+(s/defn ^:private card-query-for-tag :- (s/maybe (s/cond-pre su/Map (s/eq i/no-value)))
+  "Returns the native query for the `:card` referenced by the given tag."
+  [tag :- TagParam, _params :- (s/maybe [i/ParamValue])]
+  (when-let [card-id (:card tag)]
+    (when-let [query (db/select-one-field :dataset_query Card :id card-id)]
```

Comment:
In what case would this be false, ie. we have an id, but it doesn't map to an actual card?

Reply:
`card-id` isn't validated anywhere else on the back-end, so it may be invalid.

Reply:
Fair, but wouldn't it be then helpful to surface this error to the user, rather than silently failing?

Reply:
You're right.

Reply:
I left this as-is, but added validation [here](https://github.com/metabase/metabase/pull/11835/files#diff-9ba9c64872dacaa325097df8bcbca08bR27), because invalid card IDs trigger ["invalid database" errors](https://github.com/metabase/metabase/pull/11835/files#diff-9ba9c64872dacaa325097df8bcbca08bR42) before reaching this point.